src/agent/onefuzz-agent/src/tasks/coverage/generic.rs

```diff
@@ -362,6 +364,13 @@ impl<'a> TaskContext<'a> {
             .await
             .with_context(|| format!("writing source coverage to {}", path.display()))?;
 
+        let path = self.config.coverage.local_path.join(COBERTURA_COVERAGE_FILE);
+        let cobertura_source_coverage = cobertura(src_coverage)?;
+        let text = &cobertura_source_coverage;
```

Comment:
You should be able to just use `&cobertura_source_coverage` in the next line, if you like.

Reply:
I like separating them out- is it bad design if I do?

Reply:
Nope, not necessarily. There's a lot of subjectivity.

Here's the rationale of why I would personally choose not to do this. 
- The value we are constructing and naming is just borrow of the value `cobertura_source_coverage`. It truly is just a restricted pointer to the original thing. There's no complex, expensive, tricky, or _fallible_ conversion happening, either. Dedicating a new variable name (and a new line) adds one more thing to keep in our heads when we read ("what is the meaning of the variable `text`?"), without really giving us much for maintaining that distinction. 
- Furthermore, the syntactic weight (taking up an entire line) makes it seem like something important is going on, especially because introducing a new variable for this scenario is uncommon in Rust. But that's not true: we're just doing a regular old borrow, then immediately using it exactly once. So in my mind, the intent is made clearer (at a lower cognitive cost) just by constructing the reference exactly when we pass it to the function.
- The local name `text` is also much less descriptive than `cobertura_source_coverage`. That could be fixed, but what would we rename it to? The meaning of this variable would be best described by something like `cobertura_source_coverage_ref`. But at that point, we could just call it `&cobertura_source_coverage`, which expresses the same idea using the syntax dedicated for exactly that purpose. It also keeps our borrow duration as short as possible, which is always what we want.